---
title: "Assignment1_127_kim"
output:
  html_document:
    df_print: paged
  pdf_document: default
---
# Investigating the difference between Taiwan cohort(Chen et al., 2020) and CPTAC cohort(Gillette et al., 2020) patients' gene expression on transcriptome and proteome level.
 
## Index

#### *1. Introduction*
  1.1. Overview
  
  1.2. Detailed objectives


#### *2. Preparing the data*
  2.1. Load the data
  
  2.2. Cut out and merge the data
  
   2.2.1. TW data
    
   2.2.2. CPTAC data
  
  
#### *3. Visualization*
  3.1. Make tidier
  
  3.2. Plots
  
   3.2.1. Question 1
    
   3.2.2. Question 2
    
   3.2.3. Question 3


#### *4. Discussion*
  4.1. Summarize and discuss
  
  4.2. Imperfections in data munging
  
  4.3. Suggestions for further investigations


#### *5. Resource*

 
 
## 1. Introduction

### 1.1. Overview

Chen's research was based on Taiwanese LUAD patients' samples, and the cohort showed its unique patterns, including high proportion of never-smokers, female, early stage LUAD, and many more. Comparing to western region biased TCGA cohort, the research indicated the unique trends shown from each cohorts. 

Here, I am going to investigate the different trends of gene expression between Taiwan cohort and another cohort, on transcriptome and proteome level. To compare with Taiwan cohort(Chen et al., 2020), I selected the most recent and multi-national LUAD research(Gillette et al., 2020) in NCI(National Cancer Institute) portal which used CPTAC(Clinical Proteomic Tumor Analysis Consortium) project data. 

Since the CPTAC cohort data includes multi-national patients, it would show the global trend of gene expressions, and therefore comparing it with Taiwan cohort will identify Taiwan-specific characteristics to overall worldwide trend.

### 1.2. Detailed objectives

First, I will examine if 4 unique characteristics of Taiwan cohort mentioned in Chen's research - high proportion of nonsmokers, EGFR mutations, females, early stage patients - are also found in CPTAC multi-national cohort.

Next, I will compare RNA/protein-level gene expressions of the 7 important 'cancer related genes'(TP53, KRAS, STK11, EGFR, RB1, KEAP1, BRAF - addressed in Gillette's research) between two cohorts. These genes are significantly mutated genes with Benjamini Hochberg (BH) FDR < 0.01, visualized with the oncoplot which depicts mutually exclusive driver oncogene somatic mutations(Figure 1D in Gillette's research).

Then I will figure out if the gene expression of 7 genes correlate with the 4 properties(smoking status, EGFR mutation, gender, LUAD stage) in each cohort by ANOVA, and discuss what the results mean.

These objectives can be summarized into 3 questions :

**Q1. Do other countries on multi-national cohort also show 4 Taiwan cohort-specific characteristics(high percentage of nonsmokers, EGFR mutations, female, early stage)?**

**Q2. Compare gene expressions of 'the 7 cancer related genes'(TP53, KRAS, STK11, EGFR, RB1, KEAP1, BRAF) by cohorts.**

**Q3. Do gene expression differences relate to Taiwan cohort-specific characteristics?**



## 2. Preparing the data

Let's load the data from both of the researches and cut out any unnecessary entries from the dataframes. Also, from now on, 'TW' refers to the Taiwan cohort research(Chen et al., 2020), and 'CPTAC' refers to the CPTAC cohort research(Gillette et al., 2020).

### 2.1. Load the data
```{r}
#Load the packages and data
library(readxl)
library(ggplot2)
library(dplyr)
library(tidyverse)
library(janitor)
library(magrittr)
library(cowplot)

#Load Taiwan cohort's clinical data(patient_info), RNA expression data(RNA), and protein expression data(Protein).
d_TW_patient_info_0 = read_excel('./TW-mmc1.xlsx', sheet=2, na="NA")
d_TW_RNA_0 = read_excel('./TW-mmc1.xlsx', sheet=5, na="NA")
d_TW_Protein_0 = read_excel('./TW-mmc1.xlsx', sheet=6, na="NA")

#Load the data for CPTAC cohort.
d_CPTAC_patient_info_0 = read_excel('./CPTAC-mmc1.xlsx', sheet=2, na="NA")
d_CPTAC_RNA_0 = read_excel('./CPTAC-mmc2.xlsx', sheet=6, na="NA", skip = 2)
d_CPTAC_Protein_0 = read_excel('./CPTAC-mmc3.xlsx', sheet=2, na="NA", skip = 2)

```

### 2.2. Cut out and merge the data

We will cut out redundant information and merge the necessary information to prepare the data for visualization. Here, we use the 25 'cancer related genes' from the Cancer Gene Census (COSMIC) portal, which addressed in Chen's research(Table S1J and Figure 1A) and 7 cancer related genes addressed in Gillette's research.
```{r}
#Load the gene list from table S1J in TW research.
d_25_gene_list = read_excel('./TW-mmc1.xlsx', sheet=11, na="NA")

#These are 25 cancer related genes, selected by COSMIC. 
gene_list_1 <- d_25_gene_list %>% 
  select(Hugo_Symbol) %>% 
  pull()

#Include 6 genes, selected in Gillette's research
gene_list <- c("STK11", "KEAP1", "BRAF", gene_list_1)

#Load mutation data for TW cohort, and filter 7 genes
d_7_gene_mut = read_excel('./TW-mmc1.xlsx', sheet=4, na="NA", skip = 1)
d_7_gene_mut_1 <- d_7_gene_mut %>% 
  select(-c(2:3)) %>% 
  filter(Gene %in% c("TP53", "KRAS", "STK11", "KEAP1", "BRAF", "RB1", "EGFR")) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate(TP53.mutation.status = ifelse(is.na(TP53), "0", "1")) %>%
  mutate(KRAS.mutation.status = ifelse(is.na(KRAS), "0", "1")) %>%
  mutate(STK11.mutation.status = ifelse(is.na(STK11), "0", "1")) %>%
  mutate(RB1.mutation.status = ifelse(is.na(RB1), "0", "1")) %>%
  mutate(BRAF.mutation.status = ifelse(is.na(BRAF), "0", "1")) %>%
  mutate(KEAP1.mutation.status = ifelse(is.na(KEAP1), "0", "1")) %>%
  mutate(EGFR.mutation.status = ifelse(is.na(EGFR), "0", "1")) %>%
  rename(TP53.mutation = TP53) %>%
  rename(KRAS.mutation = KRAS) %>%
  rename(STK11.mutation = STK11) %>%
  rename(RB1.mutation = RB1) %>%
  rename(BRAF.mutation = BRAF) %>%
  rename(KEAP1.mutation = KEAP1) %>%
  rename(EGFR.mutation = EGFR)

"7_gene_list" <- c("TP53", "KRAS", "STK11", "KEAP1", "BRAF", "RB1", "EGFR")

```

#### 2.2.1. TW data

First, we will only select 25 cancer related genes from RNA, protein data frame. 
Next, we will merge the patient information(gender, age, smoking status, cancer stage, EGFR status) to RNA and protein expression data. We also flip the rows and columns around, so each rows will indicate patients, and the newly added information will become new columns.
```{r}
d_TW_patient_info <- d_TW_patient_info_0 %>%
  select(-c(Proteome_Batch, `Histology Type`, `Primary Tumor Location`, EGFR_Status))

# RNA data
d_TW_RNA <- d_TW_RNA_0 %>% 
  filter(gene %in% gene_list) %>% 
  select(-c(ensembl_gene_id, Median)) %>%
  remove_rownames() %>%
  column_to_rownames(var = 'gene') %>%
  t() %>%
  merge(d_TW_patient_info, by.x = 0, by.y = "ID") %>% #Merge data
  merge(d_7_gene_mut_1, by.x = "Row.names", by.y = 0) %>%
  rename(ID = Row.names) %>%
  mutate(Country.of.Origin = "Taiwan")
  
# Protein data
d_TW_Protein <- d_TW_Protein_0 %>% 
  filter(Gene %in% gene_list) %>% 
  select(-c(Accession, Protein)) %>%
  remove_rownames() %>%
  column_to_rownames(var = 'Gene') %>%
  t() %>%
  merge(d_TW_patient_info, by.x = 0, by.y = "ID") %>% #Merge data
  merge(d_7_gene_mut_1, by.x = "Row.names", by.y = 0) %>%
  rename(ID = Row.names) %>%
  mutate(Country.of.Origin = "Taiwan")

#Note: 3 genes are missing in protein data, and we can check them by the code below.
setdiff(gene_list, d_TW_Protein_0 %>%
          filter(Gene %in% gene_list) %>% 
          select(Gene) %>% 
          pull())

```

#### 2.2.2. CPTAC data

First, we will cut out redundant information in CPTAC data frames, and only leave patient ID, smoking status, country of origin, age, gender, EGFR mutation status, and gene abundance information for the 25 genes. Also, in CPTAC data, it *is* normalized and log2-transformed, but the tumor and NAT sample data is split. Therefore, we have to calculate the log2T/N manually.

(1) RNA data
```{r}
#Subsetting data for 28 genes, and flip rows and columns around.
d_CPTAC_RNA <- d_CPTAC_RNA_0 %>%
  filter(geneSymbol %in% gene_list | id...1 == "Type") %>%
  select(-c(id...1, gene_id, gene_type, length, id...6)) %>%
  remove_rownames() %>%
  column_to_rownames(var = 'geneSymbol') %>%
  t() %>%
  as.data.frame()

#Reorder the row names numerically, but first remove the characters.
row.names(d_CPTAC_RNA) <- row.names(d_CPTAC_RNA) %<>% 
  gsub("C3L.", "",.) %>%
  gsub("C3N.", "",.)

#Simply subtract NAT from Tumor to get log2T/N(Values are already log2-transformed). First divide into Tumor and NAT data frames, and arrange rows numerically.
d1 <- d_CPTAC_RNA %>% 
  filter(na == "Tumor") %>% 
  select(-na)

d1$index <- as.numeric(row.names(d1))
d1 <- d1[order(d1$index), ]

d2 <- d_CPTAC_RNA %>% 
  filter(na == "NAT") %>% 
  select(-na)

row.names(d2) <- gsub("\\.N", "", row.names(d2)) #Remove the character ".N"
d2$index <- as.numeric(row.names(d2))
d2 <- d2[order(d2$index), ] %>% 
  select(-index)

#Delete 9 rows that do not overlap in d1(110 rows) and d2(101 rows)
d1 <- d1 %>% 
  subset(rownames(d1) %in% intersect(row.names(d1), row.names(d2))) %>% 
  select(-index)

#Subtract 2 data frames to get log2T/N values, but first convert all the non-numeric values into numeric.
d1[] <- lapply(d1, function(x) as.numeric(as.character(x)))
d2[] <- lapply(d2, function(x) as.numeric(as.character(x)))
d_CPTAC_RNA <- d1-d2

#Merge patient_info.
d_CPTAC_patient_info_0 %<>% 
  filter(Type == "Tumor") %>% 
  select(Sample.ID, Smoking.Status, Stage, Country.of.Origin, Age, Gender, c(46:52), c(37:43))

d_CPTAC_patient_info_0$Sample.ID %<>% 
  gsub("C3L.", "", .) %>% 
  gsub("C3N.", "", .)

d_CPTAC_RNA <- merge(x = d_CPTAC_RNA, y = d_CPTAC_patient_info_0, by.x = 0, by.y = "Sample.ID")

#Mutation status columns should be 'character' vectors.

d_CPTAC_RNA$TP53.mutation.status %<>% as.character()
d_CPTAC_RNA$EGFR.mutation.status %<>% as.character()
d_CPTAC_RNA$STK11.mutation.status %<>% as.character()
d_CPTAC_RNA$RB1.mutation.status %<>% as.character()
d_CPTAC_RNA$BRAF.mutation.status %<>% as.character()
d_CPTAC_RNA$KRAS.mutation.status %<>% as.character()
d_CPTAC_RNA$KEAP1.mutation.status %<>% as.character()

```

(2) Protein data

Repeat the process for protein data.
```{r}
d_CPTAC_Protein <- d_CPTAC_Protein_0 %>%
  filter(GeneSymbol %in% gene_list | id...1 == "Type") %>%
  select(-c(1:15), -c(id...17)) %>% #16th and 17th values in GeneSymbol column are identical(both are SLC34A2), because they make two different isoforms(a, b). Let's label them.
  mutate(GeneSymbol = replace(GeneSymbol, C3L.00263 == -3.7624, 'SLC34A2-a'), GeneSymbol = replace(GeneSymbol, C3L.01890.N == 1.1266, 'SLC34A2-b')) %>% 
  remove_rownames() %>%
  column_to_rownames(var = 'GeneSymbol') %>%
  t() %>% 
  as.data.frame()

#Divide into Tumor and NAT data frames, and remove the character ".N".

d3 <- d_CPTAC_Protein %>% 
  filter(na == "Tumor") %>% 
  select(-na)

d4 <- d_CPTAC_Protein %>% 
  filter(na == "NAT") %>% 
  select(-na)

#Delete 9 rows that do not overlap in d1(110 rows) and d2(101 rows).

row.names(d4) <- gsub("\\.N", "", row.names(d4)) 

d3 <- d3 %>% 
  subset(rownames(d3) %in% intersect(row.names(d3), row.names(d4)))

#Subtract to get log2T/N values.
d3[] <- lapply(d3, function(x) as.numeric(as.character(x)))
d4[] <- lapply(d4, function(x) as.numeric(as.character(x)))
d_CPTAC_Protein <- d3 - d4


#Merge with patient_info.
row.names(d_CPTAC_Protein) <- row.names(d_CPTAC_Protein) %<>% 
  gsub("C3L.", "",.) %>%
  gsub("C3N.", "",.)

d_CPTAC_Protein <- merge(x = d_CPTAC_Protein, y = d_CPTAC_patient_info_0, by.x = 0, by.y = "Sample.ID")

#Mutation status columns should be 'character' vectors.

d_CPTAC_Protein$TP53.mutation.status %<>% as.character()
d_CPTAC_Protein$EGFR.mutation.status %<>% as.character()
d_CPTAC_Protein$STK11.mutation.status %<>% as.character()
d_CPTAC_Protein$RB1.mutation.status %<>% as.character()
d_CPTAC_Protein$BRAF.mutation.status %<>% as.character()
d_CPTAC_Protein$KRAS.mutation.status %<>% as.character()
d_CPTAC_Protein$KEAP1.mutation.status %<>% as.character()

```


## 3. Visualization

### 3.1. Make tidier

25 genes are all in the columns and it's too long. Make it tidier for visualization by pivoting. Label tidy data as '_T'.
```{r}
#Make gene list of protein data (3 genes missing) for pivoting.
gene_list_p_TW <- intersect(gene_list,  d_TW_Protein_0 %>%
                                 filter(Gene %in% gene_list) %>% 
                                 select(Gene) %>% 
                                 pull())

#Make gene list of protein data for pivoting.
gene_list_p_CPTAC <- d_CPTAC_Protein %>% select(-c(Row.names, 24:42)) %>% colnames()

#Make tidier.
d_TW_RNA_T <- d_TW_RNA %>% 
  pivot_longer(gene_list, names_to = "Gene", values_to = "Log2TN") %>% 
  rename(Smoking.Status = `Smoking Status`)

d_TW_Protein_T <- d_TW_Protein %>% 
  pivot_longer(gene_list_p_TW, names_to = "Gene", values_to = "Log2TN") %>% 
  rename(Smoking.Status = `Smoking Status`)

d_CPTAC_RNA_T <- d_CPTAC_RNA %>% 
  pivot_longer(gene_list, names_to = "Gene", values_to = "Log2TN") %>%
  rename(ID = Row.names)

d_CPTAC_Protein_T <- d_CPTAC_Protein %>% 
  pivot_longer(gene_list_p_CPTAC, names_to = "Gene", values_to = "Log2TN") %>%
  rename(ID = Row.names)

#Bind them together for easy handling. Add a column to distinguish them.
d_RNA <- bind_rows(d_TW_RNA_T, d_CPTAC_RNA_T)

d_RNA %<>% mutate(Cohort = ifelse(Country.of.Origin == "Taiwan", "Taiwan", "Multi-nation"))

d_Protein <- bind_rows(d_TW_Protein_T, d_CPTAC_Protein_T)

d_Protein %<>% mutate(Cohort = ifelse(Country.of.Origin == "Taiwan", "Taiwan", "Multi-nation"))

#Modify some column(gender, smoking status, stage) values into same formats. Ex) Gender = Female, Male, female, male --> female, male)

d_RNA$Gender %<>% 
  gsub("Female", "female", .) %>% 
  gsub("Male", "male", .)

d_RNA$Smoking.Status %<>% 
  gsub("non-smoker", "Nonsmoke", .) %>% 
  gsub("smoker", "Current_Smoker", .) %>%
  gsub("Ex-Current_Smoker", "Ex_Smoker", .)

d_RNA$Stage %<>% 
  gsub("IIIA", "3A", .) %>% 
  gsub("IIA", "2A", .) %>% 
  gsub("IA", "1A", .) %>% 
  gsub("IIIB", "3B", .) %>% 
  gsub("IIB", "2B", .) %>% 
  gsub("IB", "1B", .) %>% 
  gsub("IV", "4", .)


d_Protein$Gender %<>% 
  gsub("Female", "female", .) %>% 
  gsub("Male", "male", .)

d_Protein$Smoking.Status %<>% 
  gsub("non-smoker", "Non_smoke", .) %>% 
  gsub("smoker", "Current_Smoker", .) %>%
  gsub("Ex-Current_Smoker", "Ex_Smoker", .)


d_Protein$Stage %<>% 
  gsub("IIIA", "3A", .) %>% 
  gsub("IIA", "2A", .) %>% 
  gsub("IA", "1A", .) %>% 
  gsub("IIIB", "3B", .) %>% 
  gsub("IIB", "2B", .) %>% 
  gsub("IB", "1B", .) %>% 
  gsub("IV", "4", .)

#Let's add a column for 'region of origin' for later use.

d_RNA %<>% 
  mutate(Region.of.Origin = ifelse(Country.of.Origin %in% c("Taiwan", "china", "vietnam"), "Asian", "Western"))

d_Protein %<>% 
  mutate(Region.of.Origin = ifelse(Country.of.Origin %in% c("Taiwan", "china", "vietnam"), "Asian", "Western"))

```

```{r}
library(pander)
#My tidy data

head(d_RNA) %>% pander()

head(d_Protein) %>% pander()
```


### 3.2. Plots

#### 3.2.1. Question 1

**Q1. Do other countries on multi-national CPTAC cohort also show 4 TW cohort-specific characteristics(high percentage of nonsmokers, EGFR mutations, female, early stage)?**

```{r}
library(ggridges)
library(ggtext)
library(RColorBrewer)
library(ggrepel)
library(data.table)

#Cohort-wise comparison of :

#1. Gender
A <- d_RNA %>%
  ggplot(aes(Gender, fill = Cohort)) + 
  geom_bar(show.legend = FALSE) +
  geom_text(aes(label=..count..),stat="count",position=position_stack(0.5),size=3) +
  theme(axis.text.y = element_blank(),
        axis.ticks.y = element_blank(),
        text = element_text(size = 9)) + 
  theme(legend.position="none")

#2. Smoking status
B <- d_RNA %>%
 ggplot(aes(Smoking.Status, fill = Cohort)) + 
  geom_bar(show.legend = FALSE) +
  geom_text(aes(label=..count..),stat="count",position=position_stack(0.5),size=3) +
  theme(axis.text.y = element_blank(),
        axis.ticks.y = element_blank(),
        text = element_text(size = 9)) + 
  theme(legend.position="none")

#3. LUAD stage
C <- d_RNA %>%
  ggplot(aes(Stage, fill = Cohort)) + 
  geom_bar(show.legend = FALSE) +
  geom_text(aes(label=..count..),stat="count",position=position_stack(0.5),size=3) +
  theme(axis.text.y = element_blank(),
        axis.ticks.y = element_blank(),
        text = element_text(size = 9)) + 
  theme(legend.position="none")

#4. EGFR mutation status
D <- d_RNA %>%
  ggplot(aes(EGFR.mutation.status, fill = Cohort)) + 
  geom_bar() +
  geom_text(aes(label=..count..),stat="count",position=position_stack(0.5),size=3) +
  theme(text = element_text(size = 9),
        axis.text.y = element_blank(),
        axis.ticks.y = element_blank())


#Title
title <- ggdraw() + 
  draw_label("Taiwan-specific characteristics Compared by Cohorts",
             fontface = 'bold', x = 0, hjust = 0) +
  theme(plot.margin = margin(0, 0, 0, 7))

#Plot
plot_row1 <- plot_grid(A, B, C, D,
                       labels = c("E", "F", "G", "H"),
                       ncol = 2)

plot1 <- plot_grid(title, plot_row1,
                   ncol = 1,
                   rel_heights = c(0.1, 1))
plot1
```


As we can see, three of the four 'Taiwan-specific' characteristics designated in Chen's research - high percentage of female, nonsmokers and EGFR mutations - do not appear globally in multi-nation CPTAC cohort. However, in both cohorts, early stage patients are dominant.
To check if these properties are in fact Taiwan-specific or *Asian*-specific, plots below show the 4 characteristics by region of origin(Asian and Western) in CPTAC cohort.

```{r}
#Region-wise comparison within CPTAC cohort

#1. Gender
E <- d_RNA %>% 
  filter(Cohort == "Multi-nation") %>%
  ggplot(aes(Gender, fill = Region.of.Origin)) + 
  scale_fill_manual("legend", values = c("Asian" = "orange", "Western" = "green")) +
  geom_bar(show.legend = FALSE) +
  geom_text(aes(label=..count..),stat="count",position=position_stack(0.5),size=3) +
  theme(axis.text.y = element_blank(),
        axis.ticks.y = element_blank(),
        text = element_text(size = 9)) + 
  theme(legend.position="none")

#2. Smoking status
F <- d_RNA %>%
  filter(Cohort == "Multi-nation") %>%
  ggplot(aes(Smoking.Status, fill = Region.of.Origin)) + 
  scale_fill_manual("legend", values = c("Asian" = "orange", "Western" = "green")) +
  geom_bar(show.legend = FALSE) +
  geom_text(aes(label=..count..),stat="count",position=position_stack(0.5),size=3) +
  theme(axis.text.y = element_blank(),
        axis.ticks.y = element_blank(),
        text = element_text(size = 9)) + 
  theme(legend.position="none")

#3. LUAD stage
G <- d_RNA %>%
  filter(Cohort == "Multi-nation") %>%
  ggplot(aes(Stage, fill = Region.of.Origin)) + 
  scale_fill_manual("legend", values = c("Asian" = "orange", "Western" = "green")) +
  geom_bar(show.legend = FALSE) +
  geom_text(aes(label=..count..),stat="count",position=position_stack(0.5),size=3) +
  theme(axis.text.y = element_blank(),
        axis.ticks.y = element_blank(),
        text = element_text(size = 9)) + 
  theme(legend.position="none")

#4. EGFR mutation status
H <- d_RNA %>%
  filter(Cohort == "Multi-nation") %>%
  ggplot(aes(EGFR.mutation.status, fill = Region.of.Origin)) + 
  scale_fill_manual("legend", values = c("Asian" = "orange", "Western" = "green")) +
  geom_bar() +
  geom_text(aes(label=..count..),stat="count",position=position_stack(0.5),size=3) +
  theme(axis.text.y = element_blank(),
        axis.ticks.y = element_blank(),
        text = element_text(size = 9))

#Title
title <- ggdraw() + 
  draw_label("Taiwan-specific characteristics within CPTAC cohort",
             fontface = 'bold', x = 0, hjust = 0) +
  theme(plot.margin = margin(0, 0, 0, 7))

#Plot
plot_row2 <- plot_grid(E, F, G, H,
                       labels = c("E", "F", "G", "H"),
                       ncol = 2)

plot2 <- plot_grid(title, plot_row2,
                   ncol = 1,
                   rel_heights = c(0.1, 1))
plot2
```


As we can see, one of four characteristics - high percentage of nonsmoker - appears to be Asian-specific in CPTAC cohort.

**Answer to question 1 :**

In overall, these two sets of plots might indicate that high percentage of female and EGFR mutations are Taiwan-specific. Nonsmoker LUAD patients are dominant in Taiwan *AND* other Asian countries. Early stage patients are dominant world-wide.
In short, 'early stage' patient characteristic is not Taiwan-specific, according to these two cohorts.


#### 3.2.2. Question 2

**Q2. Compare gene expressions of 'the 7 cancer related genes'(TP53, KRAS, STK11, EGFR, RB1, KEAP1, BRAF) by cohorts.**

To answer question 2, we can create density ridge plot to visualize the gene expression by cohorts, and conduct ANOVA to compare the distributions.

1) RNA level expression

```{r}
#Density ridges plot
I <- d_RNA %>% 
  filter(Gene %in% `7_gene_list`) %>%
  mutate(Gene = fct_reorder(.f = Gene, .x = Log2TN, .fun = mean)) %>%
  ggplot(aes(Log2TN, Gene, fill = Cohort)) +
  geom_density_ridges(alpha = 0.3) +
  xlim(-3, 3) +
  geom_vline(aes(xintercept = 0)) +
  labs(title = "RNA-level expression by Cohorts") +
  theme(title = element_text(size=15))
I
```

2) Protein level expression

```{r}
#Density ridges plot
L <- d_Protein %>% 
  filter(Gene %in% `7_gene_list`) %>%
  mutate(Gene = fct_reorder(.f = Gene, .x = Log2TN, .fun = mean)) %>%
  ggplot(aes(Log2TN, Gene, fill = Cohort)) +  
  geom_density_ridges(alpha = 0.3) +
  xlim(-3, 3) +
  geom_vline(aes(xintercept = 0)) +
  labs(title = "Protein-level expression by Cohorts") +
  theme(title = element_text(size=15))
L
```


Here, we can clearly see that density ridge plot for protein-level expression is odd. It seems that multi-nation cohort's log2T/N values range much wider than that of Taiwan's. I addressed this problem in part 4.2.

Simply looking at the plot, gene expressions seem quite different between the cohorts. However, we do not know if distributions between two cohorts differ significantly or not by merely looking at the plot, so next I conducted ANOVA test.

```{r}
library(ggpubr)
library(rstatix)
library(emmeans)

#Before ANOVA, check normality in each gene group : remove extreme outliers and label '_aov'
ex_outliers <- d_RNA %>% 
  filter(Gene %in% `7_gene_list`) %>%
  group_by(Gene) %>%
  identify_outliers(Log2TN) %>%
  as.data.frame() %>%
  filter(is.extreme == TRUE) %>%
  select(Log2TN) %>%
  pull()

d_RNA_aov <- d_RNA %>%
  filter(!Log2TN %in% c(ex_outliers)) %>%
  filter(Gene %in% `7_gene_list`)

J <- d_RNA_aov %>%
  filter(Gene %in% `7_gene_list`) %>%
  ggqqplot("Log2TN", facet.by = "Gene") +
  labs(title = "Normality check - QQ plot") +
  theme(title = element_text(size=15))
J

#ANOVA test
aov_RNA <- d_RNA_aov %>% anova_test(Log2TN ~ Gene * Cohort)
aov_RNA #Note : Significant two-way interaction between Cohorts and Genes(p < 0.05).

#Post-hoc tests
#1. Simple main effects : group the data by Gene
model <- lm(Log2TN ~ Gene * Cohort, data = d_RNA_aov)
d_RNA_aov %>%
  group_by(Gene) %>%
  anova_test(Log2TN ~ Cohort, error = model) #Note : Significant differences(p > 0.05) of 7 gene expressions (mean Log2T/N) between cohorts, except for BRAF and KEAP1.

#2. Pairwise comparisons
pwc <- d_RNA_aov %>% 
  group_by(Gene) %>%
  emmeans_test(Log2TN ~ Cohort, p.adjust.method = "bonferroni") 
pwc

#Visualize
pwc <- pwc %>% add_xy_position(x = "Gene")

K <- d_RNA_aov %>%
  mutate(Gene = fct_reorder(.f = Gene, .x = Log2TN, .fun = mean)) %>%
  ggplot(aes(x = Gene, y = Log2TN, color = Cohort)) +
  geom_boxplot() +
  labs(title = "Correlation between Cohorts per Gene",
    subtitle = get_test_label(aov_RNA, detailed = TRUE),
    caption = get_pwc_label(pwc)) +
  stat_compare_means(aes(group = Cohort), label = "p.signif") +
  theme(title = element_text(size=12),
        plot.subtitle = element_text(size=8))
K
```


**Answer to question 2 : (for RNA data)**

According to ANOVA result, there are interactions between cohorts and 7 genes on Log2T/N value(gene expression), with p < 0.0001(ANOVA test). Also, mean Log2T/N(gene expression) between two cohorts differ significantly, for five of the 7 genes(EGFR, KRAS, RB1, STK11, TP53), with p < 0.05(simple main effect, pairwise comparison).

So, we verified that 5 gene expression variation of two cohorts differ from each other to a statistically meaningful extent. Then, what makes the difference between two cohorts? This leads to question 3.


#### 3.2.3. Question 3

**Q3. Are gene expression differences related to Taiwan cohort-specific characteristics?**

To answer question 3, I created density ridge plots for 5 genes, excluding two(KEAP1, BRAF) that didn't show significant difference between cohorts. Then I conducted ANOVA test for the **three** TW cohort-specific characteristics(gender, smoking status, EGFR mutation status) verified in part 3.2.1. Question 1.

- RNA level expression

Are cohort-wise gene expression differences due to :

1) Gender?
```{r}
#Density ridges plot
M <- d_RNA %>%
 filter(Gene %in% `7_gene_list`) %>%
 filter(!Gene %in% c("KEAP1", "BRAF")) %>%
 filter(!is.na(Gender)) %>%
 mutate(Gene = fct_reorder(.f = Gene, .x = Log2TN, .fun = mean)) %>%
 ggplot(aes(Log2TN, Gene, fill = Gender)) + 
 geom_density_ridges(alpha = 0.3) +
 xlim(-3, 3) +
 geom_vline(aes(xintercept = 0)) +
 labs(subtitle="Gender") +
 theme(plot.subtitle=element_text(size=14),
 text = element_text(size = 10))
M

#ANOVA test
aov_RNA_Gender <- d_RNA_aov %>% anova_test(Log2TN ~ Gene * Gender)
aov_RNA_Gender #Note : No significant two-way interaction between Gender and Genes(p > 0.05).
```


The density plot might seem to have significant difference of Log2T/N variation between female and male. However, there are no significant interactions between gender and 7 genes, with p  > 0.05(ANOVA test).


2) Smoking status?
```{r}
#Density ridges plot
N <- d_RNA %>%
 filter(Gene %in% `7_gene_list`) %>%
 filter(!Gene %in% c("KEAP1", "BRAF")) %>%
 filter(!is.na(Smoking.Status)) %>%
 mutate(Gene = fct_reorder(.f = Gene, .x = Log2TN, .fun = mean)) %>%
 ggplot(aes(Log2TN, Gene, fill = Smoking.Status)) + 
 geom_density_ridges(alpha = 0.6) +
 scale_fill_manual(values=c("#F8766D", "#FFC425", "#00BA38")) +
 xlim(-3, 3) +
 geom_vline(aes(xintercept = 0)) +
 labs(subtitle="Smoking status") +
 theme(plot.subtitle=element_text(size=14),
 text = element_text(size = 10))
N

#ANOVA test
aov_RNA_Smoke <- d_RNA_aov %>% anova_test(Log2TN ~ Gene * Smoking.Status)
aov_RNA_Smoke #Note : Significant two-way interaction between Smoking status and Genes(p < 0.05).

#Post-hoc tests
#1. Simple main effects : group the data by Gene
model <- lm(Log2TN ~ Gene * Smoking.Status, data = d_RNA_aov)
d_RNA_aov %>%
  group_by(Gene) %>%
  anova_test(Log2TN ~ Smoking.Status, error = model) #Note : Significant differences(p > 0.05) of 7 gene expressions (mean Log2T/N) between smoking status - except for BRAF, KRAS, and RB1.

#2. Pairwise comparisons
pwc_Smoke <- d_RNA_aov %>% 
  filter(!Gene %in% c("KEAP1", "BRAF")) %>%
  group_by(Gene) %>%
  emmeans_test(Log2TN ~ Smoking.Status, p.adjust.method = "bonferroni") 
pwc_Smoke

#Visualize
pwc_Smoke <- pwc_Smoke %>% add_xy_position(x = "Gene")

O <- d_RNA_aov %>%
  filter(!is.na(Smoking.Status)) %>%
  filter(!Gene %in% c("KEAP1", "BRAF")) %>%
  ggplot(aes(x = Gene, y = Log2TN, color = Smoking.Status)) +
  geom_boxplot() +
  scale_color_manual(values=c("#F8766D", "#FFC425", "#00BA38")) +
  labs(title = "Correlation between Smoking status per Gene",
    subtitle = get_test_label(aov_RNA_Smoke, detailed = TRUE),
    caption = get_pwc_label(pwc_Smoke)) +
  stat_pvalue_manual(pwc_Smoke, label = "p.adj.signif", hide.ns = TRUE) +
  theme(title = element_text(size=12),
        plot.subtitle = element_text(size=8))
O

#Pairs with significant p-value : ordered by p-value(ascending)
pwc_Smoke %>% 
  filter(p.adj.signif != "ns") %>%
  select(c(Gene, group1, group2, p)) %>%
  arrange(p) %>%
  mutate(pair = paste(Gene, paste(group1, group2, sep = ' & '), sep = '/ ')) %>%
  select(pair)
```


The density plot seem to have significant differences of Log2T/N variation between smoking status. In fact, there are significant interactions between smoking status and 5 genes on Log2T/N value(gene expression), with p < 0.0001(ANOVA test). Also, mean Log2T/N(gene expression) between three smoking status differ significantly in three genes - EGFR, TP53, STK11 - with p < 0.05(simple main effect, pairwise comparison).

3) EGFR mutation status?
```{r}
#Density ridges plot
P <- d_RNA %>%
 filter(Gene %in% `7_gene_list`) %>%
 filter(!Gene %in% c("KEAP1", "BRAF")) %>%
 filter(!is.na(EGFR.mutation.status)) %>%
 mutate(Gene = fct_reorder(.f = Gene, .x = Log2TN, .fun = mean)) %>%
 ggplot(aes(Log2TN, Gene, fill = EGFR.mutation.status)) +  
 geom_density_ridges(alpha = 0.3) +
 scale_fill_manual(values=c("orange", "purple")) +
 xlim(-3, 3) +
 geom_vline(aes(xintercept = 0)) +
 labs(subtitle="EGFR mutation status") +
 theme(plot.subtitle=element_text(size=14),
 text = element_text(size = 10))
P

#ANOVA test
aov_RNA_EGFR <- d_RNA_aov %>% anova_test(Log2TN ~ Gene * EGFR.mutation.status)
aov_RNA_EGFR #Note : Significant two-way interaction between EGFR mutation status and Genes(p < 0.05).

#Post-hoc tests
#1. Simple main effects : group the data by Gene
model <- lm(Log2TN ~ Gene * EGFR.mutation.status, data = d_RNA_aov)
d_RNA_aov %>%
  group_by(Gene) %>%
  anova_test(Log2TN ~ EGFR.mutation.status, error = model) #Note : Significant differences(p > 0.05) of 7 gene expressions (mean Log2T/N) between smoking status - except for KEAP1, KRAS, TP53.

#2. Pairwise comparisons
pwc_EGFR <- d_RNA_aov %>% 
  filter(!Gene %in% c("KEAP1", "BRAF")) %>%
  group_by(Gene) %>%
  emmeans_test(Log2TN ~ EGFR.mutation.status, p.adjust.method = "bonferroni") 
pwc_EGFR

#Visualize
pwc_EGFR <- pwc_EGFR %>% add_xy_position(x = "Gene")

Q <- d_RNA_aov %>%
  filter(!is.na(EGFR.mutation.status)) %>%
  filter(!Gene %in% c("KEAP1", "BRAF")) %>%
  ggplot(aes(x = Gene, y = Log2TN, color = EGFR.mutation.status)) +
  geom_boxplot() +
  scale_color_manual(values=c("orange", "purple")) +
  labs(title = "Correlation between EGFR mutation status per Gene",
    subtitle = get_test_label(aov_RNA_EGFR, detailed = TRUE),
    caption = get_pwc_label(pwc_EGFR)) +
  stat_pvalue_manual(pwc_EGFR, label = "p.adj.signif", hide.ns = TRUE, ) +
  theme(title = element_text(size=12),
        plot.subtitle = element_text(size=8))
Q

```


By looking at density plot, I expected EGFR and RB1 to have statistically meaningful difference between EGFR mutation existence and nonexistence. And this time, I guessed accurately. There are significant interactions between EGFR mutation status and 5 genes on Log2T/N value(gene expression), with p < 0.0001(ANOVA test). Also, mean Log2T/N(gene expression) between EGFR mutation state differ significantly in two genes - EGFR and RB1 - with p < 0.05(simple main effect, pairwise comparison).


**Answer to question 3 : (for RNA data)**

Overall, when I draw a density ridge plot, I cannnot figure out if the variation of the plots differ from each other, just by looking at them. Therefore, we have to use a tool to calculate the difference for us, and that is ANOVA. By applying ANOVA test and its follow-up tests, I found out that none of the **5** genes - whose mean Log2T/N value(gene expression) showed significant difference between TW and CPTAC cohort - were related to gender. Yet, smoking status and EGFR mutation status were related to the difference between two cohorts. In detail, three genes(EGFR, TP53, STK11) were significantly related to mean deviations between smoking status and two(EGFR, RB1) were related to difference of mean Log2T/N depending on EGFR mutation existence.



## 4. Discussion

Combine the plots
```{r, fig.dim = c(10, 10)}
#Answer 1
plot_grid(plot1, plot2,
          ncol = 1)
#Answer 2
plot_grid(I, J, K, L,
          labels = c("A", "B", "C", "D"),
          ncol = 2)
#Answer 3
plot_grid(M, NULL, N, O, P, Q,
          labels = c("A", "", "B", "C", "D", "E"),
          ncol = 2)
```


### 4.1. Summarize and discuss

To sum up, my aim in this portfolio was to compare two different cohorts which respectively represented Taiwan and world-wide LUAD patients. Here, I focused on four Taiwan-specific characteristics that were importantly addressed in Chen's research and seven cancer related genes that were addressed in Gillette's research. I attempted to compare expressions of these seven genes between cohorts in two levels - RNA and protein -, but in the process, I realized that protein-level log2T/N data of two cohorts were in fact very different from each other(addressed in 4.2. Imperfections in data munging). Therefore I focused on analyzing the difference of RNA-level gene expression between the cohorts.

I asked three questions make a comparison in three steps. 

First question was to verify if the four properties could be the reason for differences between Taiwanese-only cohort and cohort with patients from different nationalities. Here, I chose stacked barplot to visualize the difference of gender, smoking status, LUAD stage, and EGFR mutation status between two cohorts at once. As a result, three categories showed clear discrepancy between the cohorts, whereas LUAD stage category was dominated by early stage patients in both cohorts. In addition, I compared the four properties region-wise, and found out that nonsmoker patients were dominant in Asian countries, not exclusively in Taiwan. However, this conclusion might change in the future, because if the sample size grow in the future, the new samples might represent the whole population better and the resulting distributions might change as well.

I asked second question in order to find out if the 7 genes I picked were unevenly expressed in each cohort. If gene expressions are biased by cohorts, I can inspect deeper to look for any correlation between TW cohort-specific characteristics that are biased between cohorts. Initially, I created density ridge plots and tried to find the gender, smoking status and EGFR mutation status-specific tendency in RNA expressions of genes, but it was impossible without using a calculating tool. Thus in this version of portfolio, I investigated the mean difference between cohorts using ANOVA. As a result, I found difference in mean Log2T/N between cohorts to a mathematically meaningful extent. This deviation might come from any kind of characteristics of the cohort population, but I focused on only 4 categories. The columns that I excluded - age, country of origin, mutation status of other genes and so on - might be very important in classifying patients. Indeed, properties that are relevant enough might be used as a biomarker.

Last question was to find out any relationship between mean differences that I found for each characteristics and deviations I found cohort-wise. Likewise, I first attempted to figure out any relationship between the two without any calculations, but soon utilized the tool, and found out some extent of correlation with two characteristics : smoking status and EGFR mutation status. The significant pairs of mean deviation along smoking status might attribute to variation difference between two cohorts, because nonsmoker population dominated in TW cohort whereas smoker population dominated in CPTAC cohort. This also applies to the situation of EGFR mutation status : statistically important pairs that affected mean difference along EGFR mutation status might also have played a key role in the distribution difference between two cohorts. This result might also change as the cohort size grow and other significant characteristics are found. However, as we get more samples to refer to and categorize, in a more advanced analysis, this hint might proceed and add to discovering some critical index when diagnosing LUAD patients who are distinguished by their nationality, and other relevant features.

### 4.2. Imperfections in data munging 

(related to figure 2D) To answer questions 2 and 3 on both transcriptome and proteome level, I had to include plots for both. However, when drawing plots with protein-level expression data, I found out that **log2T/N value range differed greatly** between two cohorts, as we can see here:
```{r}
range(d_TW_Protein_T$Log2TN, na.rm = TRUE)

range(d_CPTAC_Protein_T$Log2TN, na.rm = TRUE)
```

After inspecting, I found out that normalization methods differed between Chen's and Gillette's research, since TW protein data was normalized by median, and CPTAC protein data was normalized by 'two-component normalization' method. Since I could not convert two-component normalized data into median normalized data, I failed in attempting to compare protein-level expressions between two cohorts.

### 4.3. Suggestions for further investigations

For further investigations, I would attempt to start with GSEA(gene set enrichment analysis) of TW cohort and CPTAC cohort. Using gprofiler, I can create a set of genes that are most prominent in each cohorts, and find related biological pathways. Then, I could make a comparison between two cohorts. Also, I would attempt exploring more columns of the merged dataset I made, because here I did not focus on various **types** of mutations of the 25 genes I initially selected. Are there any strong relationships between certain mutation types and over-/under-expressions of genes? Are some of the most correlated genes in each cohorts involved in certain pathways? What genes and mutations of genes strongly correlate to LUAD stages? I can ask a variety of questions for exploring information that I didn't focus in this portfolio. 
I tried my best to pipe(%>%) as long as I can, but pre-processing of data would have room to improve, as I have assigned excessive number of objects merely to trim down and merge the data of only two cohorts.

## 5. Resource

Gillette, M. et al. (2020). Proteogenomic Characterization Reveals Therapeutic Vulnerabilities in Lung Adenocarcinoma. _Cell, 182(1)_, 200-225. doi:https://doi.org/10.1016/j.cell.2020.06.013

Chen, Y. et al. (2020). Proteogenomics of Non-smoking Lung Cancer in East Asia Delineates Molecular Signatures of Pathogenesis and Progression. _Cell, 182(1)_, 226,244 doi:https://doi.org/10.1016/j.cell.2020.06.012
