---
title: "Tutorial-3-TCGA"
output:
  html_document:
    df_print: paged
---

# [Tutorial] Gene expression profiles in cancer patients


# 1. Introduction

Lung cancer is one of major cancers, accounting for 2.09 million deaths out of the 9.6 million total cancer deaths in 2018. There are two main types of lung cancer: small cell lung carcinoma (SCLC) and non-small cell lung carcinoma (NSCLC). NSCLC is responsible for 85–90% of lung cancer cases and its two largest subtypes are lung adenocarcinoma (LUAD) and lung squamous cell carcinoma (LUSC).

LUAD develops in the periphery of the lungs and may be associated with smoking, but is the most common lung cancer type among non-smokers. In contrast, LUSC accounts for 25–30% of all total lung cancer cases while LUAD accounts for 40% of all total lung cancer cases. LUSC are likely to be found in the middle of the lungs and is associated with smoking.

In this tutorial, we will explore the sample dataset of LUSC and LUAD from the The Cancer Genome Atlas (TCGA), a public resource for the genomic dataset. Here we use two types of lung cancers - LUAD and LUSC and will examine which information we can use for genomic analyses of lung cancers.

# 2. Obtain the gene expression profile dataset 

To access the gene expression data, there are two ways:

1. Formal way (but slow): you can download the data as described in 2.1
2. Simple ways: get the file from my dropbox link where I downloaded and save to R object as described in 2.2

## 2.1. Download the data from GDC portal

You will need to install the TCGAbiolinks package to obtain the lung cancer gene expression profiles from TCGA.
```{r}
library(tidyverse)

# Install the TCGAbiolinks package if necessary
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("TCGAbiolinks")

library(TCGAbiolinks)
```


Then, Obtain the TCGA dataset from the GDC portal. You first download the gene expression dataset for Lung Adenocarcinoma (LUAD).
```{r}
#Let's skip this formal method to save the pages!

#query <- GDCquery(project = "TCGA-LUAD",
#                  data.category = "Transcriptome Profiling",
#                  data.type = "Gene Expression Quantification", 
#                 workflow.type = "HTSeq - FPKM-UQ")
#GDCdownload(query)
#d_luad0 <- GDCprepare(query)
#d_luad = as.data.frame(d_luad0@colData)
```


Now we are downloading the dataset for Lung Squamous Cell Carcinoma (LUSC).
```{r}
#query <- GDCquery(project = "TCGA-LUSC",
#                  data.category = "Transcriptome Profiling",
#                  data.type = "Gene Expression Quantification", 
#                  workflow.type = "HTSeq - FPKM-UQ")
#GDCdownload(query)
#d_lusc0 <- GDCprepare(query)
#d_lusc = as.data.frame(d_lusc0@colData)
```


For simplicity, we are choosing only genes on the chromosome 1. Please save to the Rdata into your working folder.
```{r}
#library(SummarizedExperiment)
#e_luad = assay(d_luad0)
#e_lusc = assay(d_lusc0)

#g_luad = d_luad0@rowRanges %>% as.data.frame() %>% filter(seqnames=='chr1') %>% pull(ensembl_gene_id)
#g_lusc = d_lusc0@rowRanges %>% as.data.frame() %>% filter(seqnames=='chr1') %>% pull(ensembl_gene_id)
#e_luad = e_luad[g_luad,]
#e_lusc = e_lusc[g_lusc,]

#save(d_luad, d_lusc, e_luad, e_lusc, file='data.TCGA_LUAD_LUSC.gene_expression.Rdata')
```


## 2.2. Get the file from the link

You can download the data from this link. Please save this file to the working folder. Then, load the objects to your workspace.
```{r}
load('data.TCGA_LUAD_LUSC.gene_expression.Rdata')
```


# 3. Explore the input dataset

Let’s explore which columns are provided in the LUAD dataset.
```{r}
colnames(d_luad)
```

Do we see the same columns for the LUSC dataset?
```{r}
colnames(d_lusc)
```
**Some of the columns overlap.**


## 3.1. Which tissues or samples are available for your analysis?

Lung cancer samples in the dataset are provided in tumor or normal tissues. The column shortLetterCode contains  to describe the type of tissues collected in the dataset.

TP: Primary solid Tumor
TR: Recurrent solid Tumor
NT: Solid Tissue Normal

Check which samples are included in the LUAD dataset.
```{r}
count<-dplyr::count
d_luad %>% count(shortLetterCode)
```

shortLetterCode    n
NT    59
TP    533
TR    2

Check which samples are included in the LUSC dataset.
```{r}
d_lusc %>% count(shortLetterCode)
```

shortLetterCode    n
NT    49
TP    502


Let’s make a simple bar plot to compare the number of tissues between two tissue types. We will look at the LUAD samples first.
```{r}
d_luad %>%
  count(shortLetterCode) %>% 
  ggplot(., aes(shortLetterCode, n, fill=shortLetterCode)) + 
  geom_bar(stat="identity", position=position_dodge()) 
```


Now we can plot the LUSC samples.
```{r}
d_lusc %>%
  count(shortLetterCode) %>% 
  ggplot(., aes(shortLetterCode, n, fill=shortLetterCode)) + 
  geom_bar(stat="identity", position=position_dodge())
```

 
We made two separate plots for the LUAD and LUSC dataset. It would be great if we can merge them into one plot. Here I put the code for this. It would be good practice if you comment or delete the line that you don’t understand fully, you will compare the difference between outcomes. Also, please figure out what Qs do in this plot.
```{r}
bind_rows(d_luad %>% 
            mutate(type='luad') %>% #assign the LUAD dataset as 'luad' by adding a column.
            select(type, shortLetterCode), 
          d_lusc %>% 
            mutate(type='lusc') %>%
            select(type, shortLetterCode)) %>%
  count(shortLetterCode, type) %>% 
  complete(type, shortLetterCode, fill = list(n = 0)) %>% # Q1: What is this?
  ggplot(., aes(shortLetterCode, n, fill=shortLetterCode)) + 
  geom_bar(stat="identity", position=position_dodge()) + 
  facet_wrap(~type, scales = 'free_x', ncol=5) # Q2: What is this?
```
**A1: 'complete()' fills up missing values of all the combinations of certain columns(type, shoetLetterCode), and we can assign the missing value as '0' instead of NA with 'fill'(fill = list(n = 0)). So here, because the combination of TR and type lusc was missing, we filled that missing value with '0' instead of NA.**

**A2: facet_wrap places graphs side by side like above. The graphs are put together by type(~type), and the graph is freely drawn by x axis(scale='free_x'). Because the number of columns is 5(ncol=5), we can add another 3 graphs next to the lusc bar bar graph if needed.**



## 3.2. Distribution of clinical variables

We will plot the distribution of age at diagnosis using histogram. Few things you can check: - Is the distribution continuous? - Is it following the normal distribution? - What is the scale on the x-axis? - If the distribution is stratified, which makes it?
```{r}
# Plot histogram
ggplot(d_luad, aes(age_at_diagnosis)) + geom_histogram(bins=100)
```


The x-axis is a day scale. Let’s convert it to year.
```{r}
# Change the axis
ggplot(d_luad, aes(age_at_diagnosis/365)) + geom_histogram(bins=100)
```


To create a smooth density, we use the geom_density.
```{r}
ggplot(d_luad, aes(age_at_diagnosis)) + geom_density()
```


More plot types on distribution. First we can try the plot for both histogram and density.
```{r}
# Plot both histogram and density plot
ggplot(d_luad, aes(age_at_diagnosis/365)) + 
  geom_histogram(bins=100, colour="black", fill="white") + 
  geom_density(alpha=.2, fill="#FF6666")
```

What did you miss from this?

Let’s plot a bit different version. We will replace the y-axis of the histogram with the y-axis of the density plot.
```{r}
# Histogram with density plot
ggplot(d_luad, aes(age_at_diagnosis/365)) + 
 geom_histogram(aes(y=..density..), bins=100, colour="black", fill="white")+
 geom_density(alpha=.2, fill="#FF6666") 
```

Q: Do you think whether it is good visualization for the distribution?

**I'm not sure. In default, histogram takes frequency or counts as y axis, and density plot takes density. Because the y axis must be different, I'm not sure if showing both of the plots in the same graph is meaningful and informative. Also, we know that a smooth density plot smoothens the histogram's result. By overlapping two plots, we are just watching how the computer smoothens the histogram's result. I don't think we have to stack those two plots together.**



From the plot above, we still found the bump around the center. What would contribute to this stratification? Let’s look at some variables from other columns. First we can try information from the gender column.
```{r}
ggplot(d_luad, aes(age_at_diagnosis/365, fill=gender)) + geom_histogram(bins=100)
```


Histogram would be okay but not best visualization. What else we can try?
```{r}
ggplot(d_luad, aes(gender, age_at_diagnosis/365, fill=gender)) + 
  geom_boxplot()
```


Can you tell the difference between boxplot and density plot? We can try a violin plot.
```{r}
ggplot(d_luad, aes(gender, age_at_diagnosis/365, fill=gender)) + 
  geom_violin()
```


Can you tell the difference between boxplot and violin plot?
```{r}
# ggplot(d_luad, aes(gender, age_at_diagnosis/365, fill=gender)) +
#   geom_boxplot() + 
#   geom_violin()

# ggplot(d_luad, aes(gender, age_at_diagnosis/365, fill=gender)) +
#   geom_violin() + 
#   geom_boxplot()

# ggplot(d_luad, aes(gender, age_at_diagnosis/365)) +
#   geom_violin(aes(fill=gender)) + 
#   geom_boxplot(fill='white')
```

```{r}

ggplot(d_luad, aes(gender, age_at_diagnosis/365)) +
  geom_violin(aes(fill=gender)) + 
  geom_boxplot(fill='white', width=0.25)

ggplot(d_luad, aes(gender, age_at_diagnosis/365)) + 
  geom_violin(aes(fill=gender))  + 
  geom_boxplot(fill='white') + 
  geom_dotplot(binaxis='y', stackdir='center', dotsize=1, alpha=0.5)
#dotplot : each dot as bin width, bin along y axis, direction to stack is 'center'. 

```

```{r}
# Try different options from the manual
# http://www.sthda.com/english/wiki/ggplot2-violin-plot-quick-start-guide-r-software-and-data-visualization

#Examples:
p1 <- ggplot(d_luad, aes(x=gender, y=age_at_diagnosis/365)) + geom_violin(aes(fill=gender))
#We can try jittered plot instead of dotplot, and briefly see the distribution.
p1 + geom_jitter(shape=16, position=position_jitter(0.1))
```

Do you think gender is the reason? If not, how about tumor stages?
```{r}
ggplot(d_luad, aes(age_at_diagnosis/365)) + 
  labs(x ='Age at Diagnosis (years)', y = 'Number of LUAD patients',
       title = 'TCGA: Lung cancer adenocarcinoma') +
  geom_histogram(bins=100)
```
**Both female and male group have bimodality in their density, so I don't think it's about sex.**
 

_**More tasks for the class.**_

Q1. Add labels for the plot.
Q2. Change the color for categories.
Q3. Save to PDF file
Q4. Select the column with continuous information and plot the distribution by yourself.

```{r}
# assign p for plot
p <- ggplot(d_luad, aes(age_at_diagnosis/365)) +
  labs(x ='Age at Diagnosis (years)', y = 'Number of LUAD patients',
       title = 'TCGA: Lung cancer adenocarcinoma') +
  geom_histogram(bins=100) 

# save to PDF file
ggsave('plot.TCGA_LUAD.histogram_AgeOfDX.2021.10.27.pdf', p, width = 16, height = 9)
```

```{r}
# Let's practice the task myself.
# Q4. Plot a distribution: let's try violin plot with jittered plot.

p1 <- ggplot(d_luad, aes(x=gender, y=age_at_diagnosis/365)) +
  geom_violin(aes(fill=gender)) + #Q2. Fill the colors for 'gender' category
  geom_jitter(shape=16, position=position_jitter(0.1)) +
  labs(x='Gender', y='Age at Diagnosis (years)', title = 'TCGA: Lung cancer adenocarcinoma') #Q1. Add labels

# Save this as a PDF file
ggsave('plot.TCGA_LUAD.violin_and_jitter_AgeOfDX.2021.10.27.pdf', p1, width = 16, height = 9)
```

## 3.3. Tumor stages of lung cancers

First information you might want to explore is staging cancers. Different types of staging systems are used for different types of cancer. You can read further information in general staging rules (https://www.nhs.uk/common-health-questions/operations-tests-and-procedures/what-do-cancer-stages-and-grades-mean/), or specifics in lung cancers(e.g. stage IA).
```{r}
# Here, we have the AJCC cancer stages in 'ajcc_pathologic_stage' column, so let's change the column name
d_luad <- d_luad %>% rename(tumor_stage = ajcc_pathologic_stage)
d_lusc <- d_lusc %>% rename(tumor_stage = ajcc_pathologic_stage)
# Count the number of tumors in the LUAD dataset
counts_tumor <- d_luad %>% count(tumor_stage)

# Try bar plot
ggplot(counts_tumor, aes(tumor_stage, n)) + geom_bar(stat="identity")
```


Now you can combine both for visualization.
```{r}
# Cancer type by tumor stages
bind_rows(d_luad %>% 
            filter(shortLetterCode == 'TP') %>% 
            mutate(type='luad') %>%
            select(type, tumor_stage), 
          d_lusc %>% 
            filter(shortLetterCode == 'TP') %>% 
            mutate(type='lusc') %>%
            select(type, tumor_stage)) %>% 
  count(type, tumor_stage) %>%
  mutate(tumor_stage = factor(tumor_stage, levels=rev(unique(tumor_stage)))) %>% # Here, 'tumor_stage' is a 'unique' character column in this specific dataframe. Thus if we want to assign the tumor stage factors' levels reversely-from stage 1 to 4-, we add 'levels=rev(unique(tumor_stage))' attribute in the factor() function.
  ggplot(aes( tumor_stage, n, fill=type)) + 
  labs(title = 'Cancer type by tumor stages',
       x = '', y='Number of samples for TCGA RNA-seq') + 
  theme_minimal() + geom_bar(stat="identity") + 
  facet_wrap(~type) + coord_flip() 
```


## 3.4. Cancer type by gender

Let’s try another information - gender. We will describe this by a bar plot like above. After plotting, which information you can read from this?

NB: I am not pretty sure why TCGA put gender, not sex in the dataset, in addition to mixed use of female/male with gender.

```{r}
# Cancer type by Sex
bind_rows(d_luad %>% 
            filter(shortLetterCode == 'TP') %>% 
            mutate(type='luad') %>%
            select(type, gender), 
          d_lusc %>% 
            filter(shortLetterCode == 'TP') %>% 
            mutate(type='lusc') %>%
            select(type, gender)) %>% 
  count(type, gender) %>%
  ggplot(aes( gender, n, fill=type)) + 
  labs(title = 'Cancer type by Sex',
       x = '', y='Number of samples for TCGA RNA-seq') + 
  theme_minimal() + geom_bar(stat="identity") + 
  facet_wrap(~type) + coord_flip() 
```


Is there difference in tumor stages by gender? Let’s check with LUSC.
```{r}
# Add gender with stages
bind_rows(d_luad %>% 
            filter(shortLetterCode == 'TP') %>% 
            mutate(type='luad') %>%
            select(type, tumor_stage, gender), 
          d_lusc %>% 
            filter(shortLetterCode == 'TP') %>% 
            mutate(type='lusc') %>%
            select(type, tumor_stage, gender)) %>% 
  count(type, tumor_stage, gender) %>% 
  complete(gender, type, tumor_stage, fill = list(n = 0)) %>%
  mutate(tumor_stage = factor(tumor_stage, levels=rev(unique(tumor_stage))), 
         gender = factor(gender)) %>%
  ggplot(., aes(tumor_stage, n, fill=gender)) + 
  geom_bar(stat="identity", position=position_dodge()) + 
  facet_wrap(~type) + coord_flip() 
```
**We see a HUGE difference between sex in LUSC data. All the values for males just largely outnumber females. It is unlikely that the sex chromosomes somehow induced more LUSC to males, so probably we see the difference because the TCGA couldn't collect the LUSC samples for females as much as males.
```{r}
d_lusc %>% count(gender)
#Obviously. The male sample size is female's tripled.
```


## 3.5. Site of biopsy
```{r}
# Cancer type by biopsy
bind_rows(d_luad %>% 
            filter(shortLetterCode == 'TP') %>% 
            mutate(type='luad') %>%
            select(type, site_of_resection_or_biopsy), 
          d_lusc %>% 
            filter(shortLetterCode == 'TP') %>% 
            mutate(type='lusc') %>%
            select(type, site_of_resection_or_biopsy)) %>% 
  count(type, site_of_resection_or_biopsy) %>%
  ggplot(aes( site_of_resection_or_biopsy, n, fill=type)) + 
  labs(title = 'Cancer type by biopsy',
       x = '', y='Number of samples for TCGA RNA-seq') + 
  theme_minimal() + geom_bar(stat="identity") + 
  facet_wrap(~type) + coord_flip() 
```


Plot the years at diagnosis by tissue or organ of origin. We would also include difference by gender.
```{r}
bind_rows(d_luad %>% 
            filter(shortLetterCode == 'TP') %>% 
            mutate(type='luad') %>%
            select(type, gender, age_at_diagnosis, tissue_or_organ_of_origin), 
          d_lusc %>% 
            filter(shortLetterCode == 'TP') %>% 
            mutate(type='lusc') %>%
            select(type, gender, age_at_diagnosis, tissue_or_organ_of_origin)) %>%
  ggplot(., aes(gender, age_at_diagnosis/365, fill=gender)) + 
  geom_boxplot() + labs(y='year at diagnosis') + 
  facet_wrap(~tissue_or_organ_of_origin)
```



